services-camel/src/test/java/edu/unc/lib/dl/services/camel/fulltext/FulltextProcessorTest.java

```diff
@@ -118,4 +118,19 @@ public void extractFromInvalidPdfTest() throws Exception {
         processor.process(exchange);
         assertFalse(finalDerivativeFile.exists());
     }
+
+    @Test
+    public void extractFulltextExceedsCharacterLimit() throws Exception {
```

Comment:
Test the error message as well? Seems like a possible regression in the future.

Reply:
Do you mean check the log message warning? Not sure we're doing that anywhere at the moment, but I think its possible
https://stackoverflow.com/questions/29076981/how-to-intercept-slf4j-with-logback-logging-via-a-junit-test
It is checking that the extracted text is truncated to the character limit though, so it seems like if some other error occurs it probably won't pass that case